actions/transactions/update.go

```diff
@@ -20,21 +21,28 @@ import (
 // @Failure 	500	"Internal Server Error - Error while updating transaction"
 // @Router		/v1/transaction [patch]
 // @Security	x-auth-xpub
+// @Deprecated
 func (a *Action) update(c *gin.Context) {
-	reqXPubID := c.GetString(auth.ParamXPubHashKey)
 
 	var requestBody UpdateTransaction
 	if err := c.Bind(&requestBody); err != nil {
 		c.JSON(http.StatusBadRequest, err.Error())
 		return
 	}
+	id := requestBody.ID
+
+	a.updateTransactionWithID(c, id, requestBody.Metadata)
+}
+
+func (a *Action) updateTransactionWithID(c *gin.Context, id string, requestMetadata engine.Metadata) {
+	reqXPubID := c.GetString(auth.ParamXPubHashKey)
 
 	// Get a transaction by ID
 	transaction, err := a.Services.SpvWalletEngine.UpdateTransactionMetadata(
 		c.Request.Context(),
 		reqXPubID,
-		requestBody.ID,
-		requestBody.Metadata,
+		id,
+		requestMetadata,
 	)
 	if err != nil {
 		c.JSON(http.StatusInternalServerError, err.Error())
```

Comment:
you should update the branch and code, because there is a new method of error handling.

Reply:
Thanks. Updated.